include/lsst/afw/geom/SphPoint.h

```diff
@@ -0,0 +1,319 @@
+         *
+         * @param vector A position whose direction will be stored as a point.
+         *               Must not be the zero vector. Need not be normalized.
+         *
+         * @throws InvalidParameterError Thrown if any element of @c vector is
+         *                               NaN or infinite, or if @c vector is
+         *                               the zero vector.
+         *
+         * @par Exception Safety:
+         * The program state shall be unchanged in the event of an exception.
+         */
+        explicit SphPoint(Point3D const & vector);
+
+        /**
+         * @brief Creates a copy of a SphPoint.
+         *
+         * @param other The point to copy.
+         *
+         * @par Exception Safety:
+         * The program state shall be unchanged in the event of an exception.
+         */
+        SphPoint(SphPoint const & other) = default;
+
+        /**
+         * Overwrites this object with the value of another SphPoint.
+         *
+         * This is the only method that can alter the state of a SphPoint after
+         * its construction, and is provided to allow in-place replacement of
+         * SphPoints where swapping is not possible.
+         *
+         * @param other The object with which to overwrite this one.
+         * @return a reference to this object.
+         *
+         * @par Exception Safety:
+         * This operator shall not throw exceptions.
+         */
+        SphPoint& operator=(SphPoint const & other) = default;
+
+        /*
+         * Accessors
+         */
+
+        /**
+         * @brief The azimuth angle of this point.
+         *
+         * If this point is at a coordinate pole, this method may return any
+         * value.
+         *
+         * @return the azimuth angle, in the interval [0, 2&pi;) radians.
+         *
+         * @par Exception Safety:
+         * This method shall not throw exceptions.
+         */
+        Angle getLongitude() const noexcept { return _longitude*radians; };
+
+        /**
+         * @brief The elevation angle of this point.
+         *
+         * @return the elevation angle, in the interval [-&pi;/2, &pi;/2] radians.
+         *
+         * @par Exception Safety:
+         * This method shall not throw exceptions.
+         */
+        Angle getLatitude() const noexcept { return _latitude*radians; };
+
+        /**
+         * @brief A vector representation of this point.
+         *
+         * The adopted convention is that <1, 0, 0> corresponds to (0, 0), that
+         * <0, 1, 0> corresponds to (&pi;/2, 0), and that <0, 0, 1> corresponds
+         * to (0, &pi;/2).
+         *
+         * @return a unit vector whose direction corresponds to this point
+         *
+         * @par Exception Safety:
+         * This method shall not throw exceptions.
+         */
+        Point3D getVector() const noexcept;
+
+        /**
+         * @brief An array-like representation of this point.
```

Comment:
Please change the brief description to make it clear that it is indexing over longitude and latitude, rather than a 3d vector. Perhaps something like "Get longitude or latitude by index"